lib/charms/mysql/v0/mysql.py

```diff
@@ -3097,6 +3121,29 @@ def kill_unencrypted_sessions(self) -> None:
             logger.exception("Failed to kill external sessions")
             raise MySQLKillSessionError
 
+    def kill_client_sessions(self) -> None:
+        """Kill non local, non system open unencrypted connections.
+
+        Raises: MySQLKillSessionError if there is an issue killing the sessions.
+        """
+        kill_connections_command = (
+            f"shell.connect('{self.server_config_user}:{self.server_config_password}@{self.instance_address}')",
+            (
+                'processes = session.run_sql("'
+                "SELECT processlist_id FROM performance_schema.threads WHERE "
+                "type = 'FOREGROUND' and connection_type is not NULL and processlist_id != CONNECTION_ID();"
+                '")'
+            ),
+            "process_id_list = [id[0] for id in processes.fetch_all()]",
+            'for process_id in process_id_list:\n  session.run_sql(f"KILL CONNECTION {process_id}")',
+        )
+
+        try:
+            self._run_mysqlsh_script("\n".join(kill_connections_command))
```

Comment:
I suspect: all clients will just reconnect :-) (expected mysql client library behaviour).
What you are searching is probably MySQL Offline mode:
https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_offline_mode
https://dev.mysql.com/blog-archive/automatic-member-fencing-with-offline_mode-in-group-replication/

Reply:
[feb3b46](https://github.com/canonical/mysql-k8s-operator/pull/444/commits/feb3b46c692dbae5635f12c9e33e10e10b0fa740)